ci-automation/vendor-testing/qemu_update.sh

```diff
@@ -3,76 +3,57 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-set -euo pipefail
-
 # Test execution script for the update payload using the previous
 # release as starting point, and doing a second update from the current
 # build to itself again.
-# This script is supposed to run in the SDK container.
-
-work_dir="$1"; shift
-arch="$1"; shift
-vernum="$1"; shift
-tapfile="$1"; shift
-
-# $@ now contains tests / test patterns to run
+# This script is supposed to be sourced by vendor test runner inside
+# the mantle container.
 
-if [ "$@" != "" ] && [ "$@" != "*" ] && [ "$@" != "cl.update.payload" ]; then
-    echo "Only cl.update.payload is supported, got '$@'"
+if [ "$*" != "" ] && [ "$*" != "*" ] && [ "$*" != "cl.update.payload" ]; then
+    echo "Only cl.update.payload is supported, got '$*'"
     exit 1
 fi
 
-source ci-automation/ci_automation_common.sh
-source sdk_lib/sdk_container_common.sh
-
-mkdir -p "${work_dir}"
-cd "${work_dir}"
-
 mkdir -p tmp/
 if [ -f tmp/flatcar_test_update.gz ] ; then
-    echo "++++ QEMU test: Using existing ${work_dir}/tmp/flatcar_test_update.gz for testing ${vernum} (${arch}) ++++"
+    echo "++++ QEMU test: Using existing ./tmp/flatcar_test_update.gz for testing ${CIA_VERNUM} (${CIA_ARCH}) ++++"
 else
-    echo "++++ QEMU test: downloading flatcar_test_update.gz for ${vernum} (${arch}) ++++"
-    copy_from_buildcache "images/${arch}/${vernum}/flatcar_test_update.gz" tmp/
+    echo "++++ QEMU test: downloading flatcar_test_update.gz for ${CIA_VERNUM} (${CIA_ARCH}) ++++"
+    copy_from_buildcache "images/${CIA_ARCH}/${CIA_VERNUM}/flatcar_test_update.gz" tmp/
 fi
 
-ON_CHANNEL="$(get_git_channel)"
-if [ "${ON_CHANNEL}" = "developer" ]; then
-    # For main/dev builds we compare to last alpha release
-    ON_CHANNEL="alpha"
-fi
-if [ "${ON_CHANNEL}" = "lts" ]; then
+if [ "${CIA_CHANNEL}" = "lts" ]; then
     echo "Updating from previous LTS is not supported yet (needs creds), fallback to Stable"
-    ON_CHANNEL="stable"
+    CIA_CHANNEL="stable"
 fi
```

Comment:
I opened another PR that removes this, in case you see git conflicts due to that being merged earlier it's ok remove this block.

Reply:
Thanks for the heads-up.